onnxscript/_pygen.py

```diff
@@ -0,0 +1,507 @@
+
+    def accept(self, visitor: "Visitor"):
+        if visitor.visit_paramdef(self) and self.type:
+            self.type.accept(visitor)
+            visitor.leave(self)
+
+    def write(self, writer: TextIO):
+        if self.unpacking:
+            writer.write("*")
+        writer.write(self.name)
+        if self.type:
+            writer.write(": ")
+            self.type.write(writer)
+        if self.default_value != None:
+            value = "None" if self.default_value == NoneTypeRef \
+                else repr(self.default_value)
+            writer.write(f" = {value}")
+
+
+class Stmt(CodegenNode, ABC):
+    pass
+
+
+class Blob(Stmt):
+    def __init__(self, blob: str):
+        self.blob = blob.strip()
+
+    def accept(self, visitor: "Visitor"):
+        if visitor.visit_blob(self):
+            visitor.leave(self)
+
+    def write(self, writer: TextIO):
+        writer.write(self.blob)
+
+
+class If(Stmt):
+    def __init__(self, condition: str, *members: Stmt):
+        self.condition = condition
+        self.members = list(members)
+
+    def accept(self, visitor: "Visitor"):
+        if visitor.visit_if(self):
+            for member in self.members:
+                member.accept(visitor)
+            visitor.leave(self)
+
+    def write(self, writer: TextIO):
+        writer.write(f"if {self.condition}:\n")
+        self.write_list(
+            writer,
+            *self.members,
+            separator="\n",
+            prefix=" " * 4)
+
+
+class FunctionDef(Stmt):
+    def __init__(
+            self,
+            name: str,
+            *params: ParamDef,
+            return_type: Optional[TypeRef] = None,
+            doc: Optional[str] = None,
+            body: Optional[str] = None):
+        self.name = name
+        self.params = list(params)
+        self.return_type = return_type
+        self.doc = doc
+        self.body = body
+
+    def accept(self, visitor: "Visitor"):
+        if visitor.visit_functiondef(self):
+            for param in self.params:
+                param.accept(visitor)
+            if self.return_type:
+                self.return_type.accept(visitor)
+            visitor.leave(self)
+
+    def write(self, writer: TextIO):
+        writer.write(f"    def {self.name}(")
+        if len(self.params) == 1:
+            self.params[0].write(writer)
+        else:
+            writer.write("\n")
+            self.write_list(
+                writer,
+                *self.params,
+                prefix=" " * 12,
+                separator=",\n")
+        writer.write(")")
+        if self.return_type:
+            writer.write(" -> ")
+            self.return_type.write(writer)
+        writer.write(":\n")
+
+        def get_doc_string():
+            doc = ""
+            if self.doc:
+                doc += self.doc + "\n\n"
+            for param in self.params:
+                if param.doc:
+                    doc += f":param {param.name}: {param.doc}\n"
```

Comment:
Maybe create docstring in google style for consistency? https://google.github.io/styleguide/pyguide.html#doc-function-raises